alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/transport/TransportIndexDestinationAction.kt

```diff
@@ -198,25 +206,42 @@ class TransportIndexDestinationAction @Inject constructor(
             val getRequest = GetRequest(ScheduledJob.SCHEDULED_JOBS_INDEX, request.destinationId)
             client.get(getRequest, object : ActionListener<GetResponse> {
                 override fun onResponse(response: GetResponse) {
-                    onGetResponse(response)
+                    if (!response.isExists) {
+                        actionListener.onFailure(AlertingException.wrap(
+                            ElasticsearchStatusException("Destination with ${request.destinationId} is not found", RestStatus.NOT_FOUND)))
+                        return
+                    }
+                    val id = response.id
+                    val version = response.version
+                    val seqNo = response.seqNo.toInt()
+                    val primaryTerm = response.primaryTerm.toInt()
+                    val xcp = XContentFactory.xContent(XContentType.JSON)
+                        .createParser(xContentRegistry, LoggingDeprecationHandler.INSTANCE, response.sourceAsString)
+                    XContentParserUtils.ensureExpectedToken(XContentParser.Token.START_OBJECT, xcp.nextToken(), xcp)
+                    XContentParserUtils.ensureExpectedToken(XContentParser.Token.FIELD_NAME, xcp.nextToken(), xcp)
+                    XContentParserUtils.ensureExpectedToken(XContentParser.Token.START_OBJECT, xcp.nextToken(), xcp)
+                    val dest = Destination.parse(xcp, id, version, seqNo, primaryTerm)
+                    onGetResponse(dest)
                 }
                 override fun onFailure(t: Exception) {
                     actionListener.onFailure(AlertingException.wrap(t))
                 }
             })
         }
 
-        private fun onGetResponse(response: GetResponse) {
-            if (!response.isExists) {
+        private fun onGetResponse(destination: Destination) {
+            val backendRoles = destination.user?.backendRoles
+            if (filterByEnabled &&
+                (backendRoles == null || user?.backendRoles == null || backendRoles.intersect(user.backendRoles).isEmpty())) {
                 actionListener.onFailure(AlertingException.wrap(
-                        ElasticsearchStatusException("Destination with ${request.destinationId} is not found", RestStatus.NOT_FOUND)))
-                return
+                    ElasticsearchStatusException("Do not have backend roles to delete destination with ${request.destinationId}",
```

Comment:
Can we make the message more generic? `Do not have permissions to delete destination with ...`

Reply:
Done